### DISCUSSION TITLE
使用svp补帧相关

### DISCUSSION THREAD

**Discussion by hooke007** [hooke007/mpv_PlayKit#114] [INFO 信息]
mpv支持svp补帧法

>[!NOTE]
>_非使用新整合包的其他mpv用户，需要结合 “**原文**” 部分（因部分文件已过时而不再维护）自行融合理解_

此功能现受 K7sfunc 包装器的多个模块例如 [SVP_LQ](https://github.com/hooke007/mpv_PlayKit/wiki/3_K7sfunc#svp_lq) 的支持。
但是 mpv-lazy-20221011 及之后的版本不再集成对应的依赖插件，需要手动获取（参见 [二楼的附件](https://github.com/hooke007/mpv_PlayKit/discussions/114#discussioncomment-6399037)）

安装说明：

0. 已部署 mpv-lazy 的标准包
1. 解压两个插件 svpflow1 svpflow2 至 vapoursynth 的[插件路径](https://www.vapoursynth.com/doc/installation.html#windows-portable)下
2. 参考 K7sfunc 的维基说明自行构建vpy脚本；或者下载仓库的示例脚本，例如 [MEMC_SVP_LQ](https://github.com/hooke007/mpv_PlayKit/blob/main/portable_config/vs/MEMC_SVP_LQ.vpy) 。vpy脚本的安置路径任意
3. 编辑 input_uosc.conf 文件，链接至对应文件即可在运行时启用（自行参考其它滤镜类的写法）。

完毕。

<br>
<br>
<br>
<br>
<br>
<br>
<br>
<br>
<br>
<br>
<br>
<br>
<br>
<br>
<br>

>[!WARNING]
>_此处开始为旧的 **原文**_

旧版的svpflow可以免manager运行并曾经在官网免费提供。
旧版懒人包已集成，mpvnet_CM 未集成，可以通过下方对应的步骤快速使用。

## 1. mpv 原版

[旧版懒人包](https://github.com/hooke007/mpv_PlayKit/releases/tag/20220903)已集成，可直接启用。
如果要基于你自己的mpv配置，参考阅读“关联”的文章。

如果需要使用自己的svp manager，按照 [【FAQ】](https://github.com/hooke007/mpv_PlayKit/wiki/0_FAQ#%E4%BD%BF%E7%94%A8%E5%8E%9F%E7%89%88svp-manager%E7%A8%8B%E5%BA%8F) 处理。

关联： [《mvtools的安装示例》](https://hooke007.github.io/unofficial/mpv_start.html#mvtools)
扩展： [《svpflow的可配置参数》](https://www.svp-team.com/wiki/Manual:SVPflow)

## 2. mpv.net_CM （已过时）

> _如果你有svp manager，那就按照 [说明](https://github.com/hooke007/mpv.net_CM#%E5%85%BC%E5%AE%B9-svp-manager) 正常使用，无需看这部分_

4.1. 下载旧版懒人包，复制其中的所有文件名中带`svpflow`的文件到mpvnet中的对应位置
  vapoursynth64/plugins/**svpflow1_vs64.dll**
  vapoursynth64/plugins/**svpflow1_vs64.dll**
  <s>portable_config/vs/**svpflow-2x.vpy**
  portable_config/vs/**svpflow-8b.vpy**
  portable_config/vs/**svpflow-Starlight.vpy**</s>
  （这三个脚本文件用[主仓库](https://github.com/hooke007/mpv_PlayKit/tree/main/portable_config/vs)中的文件 svpflow_*.vpy 代替）

2.2. 根据你的使用习惯，自行编辑 input.conf 或 mpv.conf
 _对应语法参考mvtools_

#### 2.3 BlackMicky 黑米奇版的特别处理

由于加入了自制功能和额外插件，与mpvnet_CM的组件冲突，需要参照 mpv-lazy 的FAQ彻底清理相关文件。
相关反馈： https://github.com/hooke007/mpv_PlayKit/discussions/82#discussioncomment-1937492

## 3. 可能有效的效率优化

> mpv 的 vapoursynth=file:buffered-frames:concurrent-frames
buffered-frames 设为 1 即可，对于补帧来说只需要用到两帧
concurrent-frames 默认等于逻辑处理器数量，我测下来等于 1.5 倍逻辑处理器数量时一些默认会掉帧复杂场景能处理的过来了（Haswell 4核无超线程 svp to 60，Whiskey Lake 6核开超线程 svp to 120，Vermeer 12核开超线程 rife x2 + svp 120）。不过不是很理解为什么，这里应该只是解码消耗，特别是24个逻辑处理器也有这种现象，很怪异

完整写法示例：
`--vf=vapoursynth="~~/xxxxx.vpy":buffered-frames=2:concurrent-frames=8`
或精简为 `--vf=vapoursynth="~~/xxxxx.vpy":2:8`

**Comment by Xianglos** [comment]
更新一下涉及的两个插件。
svpflow1 继承了mvt的gpl协议似乎被迫开源 http://www.svp-team.com/files/gpl/svpflow1-src.zip
svpflow2 闭源，这里有个q群友传的168无后台版，和161比起来也没有什么实际的更新（出处不知道，官方就没发过这个，怕毒就用161）

- [svpflow1_4.5.0.200.zip](https://github.com/hooke007/mpv_PlayKit/files/11996855/svpflow1_4.5.0.200.zip)
- [svpflow2_4.3.0.161.zip](https://github.com/hooke007/mpv_PlayKit/files/11996856/svpflow2_4.3.0.161.zip)
 [svpflow2_4.3.0.168.zip](https://github.com/hooke007/mpv_PlayKit/files/11996857/svpflow2_4.3.0.168.zip)

**Reply by verygoodlee** [reply]
168版出自potplayer吧的一位吧友，先后破解了`avisynth`和`vapoursynth`的两个，只提供了64位的\
对比了一下，文件哈希值是相同的\
原帖地址：\
[发个svpflow 168版，没有红框的那种](https://tieba.baidu.com/p/7704528706)
[svpflow4.3.0.168 不需要manager版](https://tieba.baidu.com/p/7708386022)

**Comment by d3xMachina** [comment]
I modified the latest version (4.6.0.250) to remove the red borders. It's for vapoursynth.

[svpflow_4.6.0.250_vapoursynth.zip](https://github.com/hooke007/mpv_PlayKit/files/12639668/svpflow_4.6.0.250.zip)

**Reply by lauyv** [reply]
Is it possible to update the 4.6.0.263 version ~~because this won't turn on nvof~~.
when nvof turn on, 250 version will flash exit

**Reply by hooke007** [reply]
nvof won't be active unless you manually set it. Which function were you using?

**Reply by lauyv** [reply]
yes, i manually set it.
It be fine when i get svpflow2 4.6.0.263 from svp official website, but border is red.
`clip = k7f.SVP_PRO(input=video_in, fps_in=container_fps, nvof=True, fps_num=2)`

**Reply by hooke007** [reply]
You mean update only?
I thought you were talking about that the current version's nvof is broken.

**Reply by lauyv** [reply]
Sry, my expression was wrong...
When nvof turn on, 250 version will flash exit, 263 version dont flash exit but with red border

**Reply by d3xMachina** [reply]
Check the last post for 263 version.

**Reply by lauyv** [reply]
it works as expected~ ty

**Comment by jiang-yuan** [comment]
这个svp开源版vs的性能是不是远不如现在的SVP4pro啊? 之前使用集成这个的lazy-mvp,开启以后感觉都在吃满CPU而没有用GPU,效率很低

**Reply by hooke007** [reply]
没有开源版。一样的东西。你都没对齐参数怎么可能效率一致。

**Reply by d3xMachina** [reply]
SVP4pro use the exact same dll. You need to set the parameters to use the GPU.

**Reply by jiang-yuan** [reply]
自己改.vs文件?

**Comment by d3xMachina** [comment]
Version 4.6.0.263 with red borders removed :

[svpflow_4.6.0.263_vapoursynth.zip](https://github.com/hooke007/mpv_PlayKit/files/15047933/svpflow_4.6.0.263_vapoursynth.zip)

**Reply by lauyv** [reply]
thx!!!!!!!

**Reply by d3xMachina** [reply]
> > Version 4.6.0.263 with red borders removed :
> > [svpflow_4.6.0.263_vapoursynth.zip](https://github.com/hooke007/mpv_PlayKit/files/15047933/svpflow_4.6.0.263_vapoursynth.zip)
> 
> 兄弟，有最新的4.7.0的无红框版本吗

Latest version of svpflow included in the SVP installer is 4.6.0.274. This should work to remove the red borders (not tested, I don't use svpflow anymore) : 
[svpflow_4.6.0.274_vapoursynth.zip](https://github.com/user-attachments/files/23752928/svpflow_4.6.0.274_vapoursynth.zip)

**Reply by d3xMachina** [reply]
> > > > Version 4.6.0.263 with red borders removed :
> > > > [svpflow_4.6.0.263_vapoursynth.zip](https://github.com/hooke007/mpv_PlayKit/files/15047933/svpflow_4.6.0.263_vapoursynth.zip)
> > > 
> > > 
> > > 兄弟，有最新的4.7.0的无红框版本吗
> > 
> > 
> > Latest version of svpflow included in the SVP installer is 4.6.0.274. This should work to remove the red borders (not tested, I don't use svpflow anymore) : [svpflow_4.6.0.274_vapoursynth.zip](https://github.com/user-attachments/files/23752928/svpflow_4.6.0.274_vapoursynth.zip)
> 
> Which one do you usually use now?

I use [BlueskyFRC](https://bluesky-soft.com/en/BlueskyFRC.html). It only works with AMD and Intel graphic cards. It gives better results than SVP on my AMD GPU (more fluidity and less artifacts) and it's easy to install.

**Reply by Leavesr** [reply]
> > > > > Version 4.6.0.263 with red borders removed :
> > > > > [svpflow_4.6.0.263_vapoursynth.zip](https://github.com/hooke007/mpv_PlayKit/files/15047933/svpflow_4.6.0.263_vapoursynth.zip)
> > > > 
> > > > 
> > > > 兄弟，有最新的4.7.0的无红框版本吗
> > > 
> > > 
> > > Latest version of svpflow included in the SVP installer is 4.6.0.274. This should work to remove the red borders (not tested, I don't use svpflow anymore) : [svpflow_4.6.0.274_vapoursynth.zip](https://github.com/user-attachments/files/23752928/svpflow_4.6.0.274_vapoursynth.zip)
> > 
> > 
> > Which one do you usually use now?
> 
> I use [BlueskyFRC](https://bluesky-soft.com/en/BlueskyFRC.html). It only works with AMD and Intel graphic cards. It gives better results than SVP on my AMD GPU (more fluidity and less artifacts) and it's easy to install.

BlueskyFRC doesn't seem to work with MPV; it appears to only work with MPC and PotPlayer.

**Comment by zaeboba** [comment]
I did everything as described, created a folder vapoursynth64/plugins/ put two files there, but nothing works. 

<details>
<summary>clic</summary>

[](https://github.com/user-attachments/assets/34b09b69-9e6b-4960-8716-5806df323eba)

</details>

**Reply by hooke007** [reply]
Always check the console/log first instead of making a useless screen recording.
> created a folder vapoursynth64/plugins/

It's the deprecated folder for vapoursynth.

**Reply by zaeboba** [reply]
![изображение](https://github.com/user-attachments/assets/81806b61-5bec-4362-9439-c8cc8e23e225)

**Reply by hooke007** [reply]
svpdll was not detected on your device. Place them into the correct dir.

**Reply by zaeboba** [reply]
this is the full version of the lazy man, which weighs 1 gigabyte. 🤷‍♂️

**Reply by hooke007** [reply]
svp was not bundled in any lazy pack.

**Reply by zaeboba** [reply]
tell me exactly what I need to download and where to put it.
I don't really understand

**Reply by hooke007** [reply]
https://www.vapoursynth.com/doc/installation.html#windows-portable

**Reply by zaeboba** [reply]
![изображение](https://github.com/user-attachments/assets/a678f7cb-96bd-41a5-8bd2-7e1a00bd60dd)
I have a portable, right.

**Reply by zaeboba** [reply]
Oh, sorry, I'm confused about the folders.  🤣

**Reply by hooke007** [reply]
It isn't my pack at all.
I don't know what you were doing on your side.
![image](https://github.com/user-attachments/assets/13e9537a-38d6-41c0-b1cd-7297a5719a2b)

**Reply by zaeboba** [reply]
it's yours, I just moved it to the portable folder. https://github.com/hooke007/mpv_PlayKit/discussions/484#discussion-6938465

**Reply by hooke007** [reply]
Can't help more if any non-standard installation.

**Reply by zaeboba** [reply]
luckily you've already helped.
